metadata-ingestion/src/datahub/ingestion/source/mode.py

```diff
@@ -900,18 +919,24 @@ def normalize_mode_query(self, query: str) -> str:
 
         return rendered_query
 
-    def construct_query_from_api_data(
+    def construct_query_or_dataset(
         self,
         report_token: str,
         query_data: dict,
         space_token: str,
         report_info: dict,
+        is_mode_dataset: bool = False,
     ) -> Iterable[MetadataWorkUnit]:
-        query_urn = self.get_dataset_urn_from_query(query_data)
+        query_urn = (
+            self.get_dataset_urn_from_query(query_data)
+            if not is_mode_dataset
+            else self.get_dataset_urn_from_query(report_info)
```

Comment:
if a report depends on multiple datasets, what happens?

Reply:
hi @hsheth2 
we first identify all the mode-datasets associated with a given report, then include these mode-datasets into the report’s dataset urns (dashboard).

After updating the report(dashboard), we process each mode-dataset individually and ensure they are added as datahub dataset.

Example: For a report containing two mode-datasets
1. BQ-Dataset 
2. NewDatasetTest
This are mode datasets added into Report name called DatasetReport_Test

<img width="1432" alt="Screenshot 2024-09-05 at 9 03 00 PM" src="https://github.com/user-attachments/assets/7b7e1a86-7e33-4994-b305-c1fc37a3b2cf">

Reply:
ok